src/integrationtest/java/com/ericsson/ei/frontend/CommonSteps.java

```diff
@@ -0,0 +1,154 @@
+        httpRequest.setHost(HOST).setPort(frontendPort).setEndpoint(endpoint);
+    }
+
+    @When("^\'(.*)\' is appended to endpoint$")
+    public void append_to_endpoint(String append) throws Throwable {
+        String endpoint = httpRequest.getEndpoint() + append;
+        httpRequest.setEndpoint(endpoint);
+    }
+
+    @When("^param key \'(.*)\' with value \'(.*)\' is added$")
+    public void add_param(String key, String value) throws Throwable {
+        httpRequest.addParam(key, value);
+    }
+
+    @When("^body is set to file \'(.*)\'$")
+    public void set_body(String filename) throws Throwable {
+        String filePath = this.getClass().getResource(BODIES_PATH + filename).getFile();
+        String fileContent = FileUtils.readFileToString(new File(filePath), "UTF-8");
+        httpRequest.addHeader("Content-type", "application/json").setBody(fileContent);
+    }
+
+    @When("^aggregation is prepared with rules file \'(.*)\' and events file \'(.*)\'$")
+    public void aggregation_is_prepared(String rulesFileName, String eventsFileName) throws Throwable {
+        String rulesPath = this.getClass().getResource(BODIES_PATH + rulesFileName).getFile();
+        String eventsPath = this.getClass().getResource(BODIES_PATH + eventsFileName).getFile();
+        String rules = FileUtils.readFileToString(new File(rulesPath), "UTF-8");
+        String events = FileUtils.readFileToString(new File(eventsPath), "UTF-8");
+        String body = new JSONObject().put("listRulesJson", new JSONArray(rules))
+                .put("listEventsJson", new JSONArray(events)).toString();
+        httpRequest.setBody(body);
+    }
+
+    @When("^username \"(\\w+)\" and password \"(\\w+)\" is used as credentials$")
+    public void with_credentials(String username, String password) throws Throwable {
+        String auth = username + ":" + password;
+        String encodedAuth = new String(Base64.encodeBase64(auth.getBytes()), "UTF-8");
+        httpRequest.addHeader("Authorization", "Basic " + encodedAuth);
+    }
+
+    @When("^request is sent$")
+    public void request_sent() throws Throwable {
+        response = httpRequest.performRequest();
+    }
+
+    @Then("^response code (\\d+) is received$")
+    public void get_response_code(int statusCode) throws Throwable {
+        LOGGER.info("Response code: {}", response.getStatusCode());
+        assertEquals(HttpStatus.valueOf(statusCode), response.getStatusCode());
+    }
+
+    @Then("^response body \'(.*)\' is received$")
+    public void get_response_body(String body) throws Throwable {
+        LOGGER.info("Response body: {}", response.getBody());
+        assertEquals(body, response.getBody());
+    }
+
+    @Then("^response body from file \'(.*)\' is received$")
+    public void get_response_body_from_file(String filename) throws Throwable {
+        String filePath = this.getClass().getResource(RESPONSES_PATH + filename).getFile();
```

Comment:
Maybe join the path and filename using some lib so that it's not dependent on writing / or without / in the end of the path.

Reply:
I don't think that is needed. The PATH constant points to a directory in resources and filename points to a file in that folder. The only time an error would occur is if someone writes a filename with a '/' at the start but that can simply be fixed by removing it.